### DISCUSSION TITLE
Provide a way for RadioComponent / CheckboxComponent to handle "checked status" for object values

### DISCUSSION THREAD

**Discussion by Cr3aHal0** [primefaces/community#1254] [PrimeNG] [answered]
Hi everyone,

there are currently issues with combobox and radio when bound to reactive form and when using object values. An exemple below : 

https://stackblitz.com/edit/wnctrn?file=src%2Fapp%2Fdemo%2Fradio-button-reactive-forms-demo.ts

As you can see, the first radio should be checked because the value if the same but it isn't checked because the reference isn't the same (since the value is a new object which has the same properties and values than the one in the provided options). 

Sure, there is a workaround such as 

```
    this.formGroup = new FormGroup({
      selectedCategory: new FormControl(this.categories[0]),
    });
```

but when working with real life examples and forms (like loading data from an API and patching the form with the provided data) such workaround feels unnecessary and tedious as dropdown and other complexe form field components already handle this kind of value.

Should/could combobox and radio be using some kind of `valueKey` input like most other components which would be used (or not when not provided) for data comparison, or some kind of identity function input for the same purpose ?

Thank you guys in advance,

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks a lot for the explanation and sharing an example. You're right it should work as it works in the dropdown or other input components. I've opened an issue for this, we'll try to address it in v17.

Issue link: https://github.com/primefaces/primeng/issues/14880
